### DISCUSSION TITLE
Cell input to mace training

### DISCUSSION THREAD

**Discussion by lamthuy** [ACEsuit/mace#410] [General]
Hi,
I am new to this area. I have a question regarding using MACE APIs to train on energy, force and stress form the MPtrij data. The data include energy, force and stress for every configuration. I read the Mace code and see that when it calculate stress, it requires information about cell. If cell is not provided as input it will uses the default cell value initialised as a zero 3x3 matrix. I am wondering if initializing with a  zero 3x3 matrix is a correct way to train with stress on the MP data? I saw in the MP data there are latice matrices, not sure if this is related to the cell input?
Thank you in advance!

**Comment by gabor1** [comment]
If a cell is not provided, the stress cannot be computed (since the stress is the derivative of the total energy with respect to the deformation of the cell). The initialisation with zero just caters for the situation when the input has no cell and no stress data (eh isolated molecules in vacuum)

**Reply by lamthuy** [reply]
Thank you for the quick response. I tried without cell information and it actually returns inf as tress. Is the cell 3x3 matrix the same as the lattice matrix  in the dataset like MPTrj?

**Reply by gabor1** [reply]
Again that's just a placeholder. Stress is not defined when there is no cell. In ASE there is another way to indicate explicitly that there are no periodic boundary conditions: pbc=False

And yes, Lattice is the same as cell.
